{{cookiecutter.distribution_name}}/setup.py

```diff
@@ -10,6 +10,13 @@
     hit = _version_re.search(f.read().decode('utf-8')).group(1)
     version = str(ast.literal_eval(hit))
 
+
+# Because q2cli uses a custom caching system to implement tab completion, the
```

Comment:
Here's some text that I adapted from [my forum post](https://forum.qiime2.org/t/plugin-development-visualizer-wont-unregister-despite-deleting-it/153/3?u=jairideout) and [@ebolyen's forum post](https://forum.qiime2.org/t/plugin-development-visualizer-wont-unregister-despite-deleting-it/153/2?u=jairideout) on this topic. I think the text is more accurate about why and when exactly this needs to happen. The current text makes it sound like the cache needs refreshing when a new plugin is installed, but this case only happens when developing a plugin, not with releases. When using the CLI with releases, the cache will auto-refresh when anything changes in the deployment.

```
NOTE: If you are testing your plugin with `q2cli` (i.e. the `qiime` command) while you are developing it, you'll need to run `qiime dev refresh-cache` to see the latest changes to your plugin reflected in the CLI. You'll need to run this command anytime you modify your plugin's interface (e.g. add/rename/remove a command or its inputs/parameters/outputs).

Another option is to set the environment variable `Q2CLIDEV=1` so that the cache is refreshed every time a command is run. This will slow down the CLI while developing because refreshing the cache is slow. However, the CLI is much faster when a user installs release versions of QIIME 2 and plugins, so this slowdown should only be apparent when *developing* a plugin.

This manual refreshing of the `q2cli` cache is necessary because it can't detect when changes are made to a plugin's code while under development (the plugin's version remains the same across code edits). This manual refreshing of the cache should only be necessary while developing a plugin; when users install QIIME 2 and your released plugin (i.e. no longer in development), the cache will automatically be updated when necessary.
```